### ISSUE TITLE
Markdown block: Parsing issues and inconsistencies

### ISSUE THREAD

**Issue by tyxla** [Automattic/jetpack#11881]
In testing how elements are recognized and parsed in our Jetpack Markdown block for Gutenberg, we've discovered several issues and inconsistencies so far. 

This is a master issue for all of them, feel free to add any others that you find.

* [ ] Footnotes / reference links differences
	* Reported by @mapk in https://github.com/Automattic/jetpack/pull/10133#issuecomment-421158443
	* Reported by @jeherve in p1HpG7-5EW-p2 #comment-27928
* [ ] Images with alts and titles not supported
	* Reported by @jeherve in p1HpG7-5EW-p2 #comment-27928
* [ ] Single empty lines are omitted
	* Reported by @Stojdza in p1HpG7-5EW-p2 #comment-27911
	* Reported by @carladoria in p58i-7sR-p2 #comment-40539
	* Reported by @reginabally in p58i-7sR-p2 #comment-40610
* [ ] Alternative heading syntax parsed incorrectly on iPad Pro
	* Reported by @eliorivero in p1HpG7-5EW-p2 #comment-27968
* [ ] Code block not properly styled
	* Reported by @carladoria in p58i-7sR-p2 #comment-40539

**Comment by chrisfromthelc** [comment]
I discovered through a support interaction that [Footnotes syntax](https://michelf.ca/projects/php-markdown/extra/#footnotes) does not work, even though we say it's supported (via Markdown Extra) [here](https://wordpress.com/support/wordpress-editor/blocks/markdown-block/#markdown-extra-and-markdown-in-jetpack).

I've tested and confirmed on a simple WordPress.com Site, an Atomic Site, and a Jetpack-connected self-hosted site to confirm.

**Comment by KokkieH** [comment]
Another report of this issue in https://wordpress.com/forums/topic/markdown-code-syntax-highlightyng-using-jetpack/

**Comment by DBJDBJ** [comment]
Guys, what is not being said here? 2+ years for NOT resolving something so important to many users. There must be a story behind this.

**Comment by jeherve** [comment]
@DBJDBJ We're not currently working on the Markdown block, but when we do come back and iterate on that block, we'll definitely be tackling some of the items on the list above.

Until then, we've marked this issue as "Good For Community". That means that we'll be happy to review any improvements and bug fixes suggested by community members like you.

**Comment by DBJDBJ** [comment]
@jeherve   What is good for me I have now done: uninstalled Jet Pack.  Actually I would very much like to move to HUGO but dbj.org is 500+ posts. It will be done.

**Comment by mxro** [comment]
Any chance on getting proper syntax highlighting for code blocks? As far as I can see, syntax highlighting still does not work even when specifying the correct language. All code in Markdown blocks is just rendered as plain text.

![image](https://user-images.githubusercontent.com/1448524/166084691-ae3a05a4-5dba-42dd-bae8-aafd8cb8f1e1.png)

**Comment by jenxi** [comment]
Any chances of adding Markdown footnotes support in Markdown blocks?

**Comment by sophiegyo** [comment]
We had a question from someone who would like to create page anchors in Markdown:
https://wordpress.com/forums/topic/markdown-and-linkable-section-headers/

They referenced this .org style guide:
https://make.wordpress.org/docs/style-guide/linking/heading-targets/

I learned that the Markdown block uses [CommonMark spec](https://spec.commonmark.org/0.30/) but this doesn't appear to allow page anchors, nor does it seem that the .org method of page anchors works in the Markdown block. I understand work's been done on the footnotes in the block; would it be viable to extend this to regular page anchors? (Or of course if there's something I'm missing about how to make those work, I'll gladly learn!)

**Comment by jeherve** [comment]
> this doesn't appear to allow page anchors

That's correct. Header anchors are not supported at the moment. You can read more about it in this discussion: this doesn't appear to allow page anchors

> They referenced this .org style guide:
> [make.wordpress.org/docs/style-guide/linking/heading-targets](https://make.wordpress.org/docs/style-guide/linking/heading-targets/)

This is about content published on make.wordpress.org, so it's a bit different. Instead, I would recommend using the option provided in the Heading block to add anchors to headings, like so:

<img width="1383" alt="Screenshot 2023-06-07 at 14 05 57" src="https://github.com/Automattic/jetpack/assets/426388/8a092d27-2a8a-45ea-a59d-675cc2cf47ba">

That requires using a mix of Markdown blocks and heading blocks, which isn't ideal, but that's one way to work around the problem today.
